flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/DeclarativeSlotPoolBridge.java

```diff
@@ -548,4 +610,13 @@ void increaseResourceRequirementsBy(ResourceCounter increment) {
     boolean isBatchSlotRequestTimeoutCheckEnabled() {
         return !isBatchSlotRequestTimeoutCheckDisabled;
     }
+
+    @VisibleForTesting
+    public void tryWaitSlotRequestIsDone() {
+        if (getDeclarativeSlotPool() instanceof DefaultDeclarativeSlotPool) {
+            final DefaultDeclarativeSlotPool slotPool =
+                    (DefaultDeclarativeSlotPool) getDeclarativeSlotPool();
+            slotPool.tryWaitSlotRequestIsDone();
+        }
+    }
```

Comment:
If last comment makes sense, could we move ` void tryWaitSlotRequestIsDone()` method to `DeclarativeSlotPoolBridgeTest`?

Only `DeclarativeSlotPoolBridgeTest` calls `tryWaitSlotRequestIsDone()`.